src/veh_type.cpp

```diff
@@ -551,10 +551,80 @@ const std::set<std::string> &vpart_info::get_categories() const
     return this->categories;
 }
 
+class turret_migration_helper
+{
+    public:
+        static translation translate_gun_to_turret( const itype &gun_type ) {
+            return to_translation( "mounted " + gun_type.name.raw );
+        }
+};
+
 void vpart_info::finalize()
 {
     DynamicDataLoader::get_instance().load_deferred( deferred );
 
+    static const auto mountable_gun_filter = []( const itype & t ) {
+        // returns true for "valid" gun items which can be mounted as vehicle turret
+
+        static const std::vector<flag_id> bad_flags {
+            flag_RELOAD_AND_SHOOT,
+            flag_PRIMITIVE_RANGED_WEAPON,
```

Comment:
`PRIMITIVE_RANGED_WEAPON` seems unnecessary here; `RELOAD_AND_SHOOT` would already exclude bows, slings and similar things, while allowing crossbows that would indeed be possible to mount on a vehicle.

Reply:
That makes sense, I'll resolve next time I get to this branch 👍